cermine-impl/src/main/java/pl/edu/icm/cermine/ContentExtractor.java

```diff
@@ -847,20 +844,122 @@ public static void main(String[] args) throws ParseException, AnalysisException,
                 printException(ex);
             } catch (TimeoutException ex) {
                 printException(ex);
+            }catch (IOException ex) {
+                printException(ex);
+            }catch (Exception ex) {
+                printException(ex);
             } finally {
                 if (extractor != null) {
+                    extractor.printProgress(total.intValue());
                     extractor.removeTimeout();
                 }
-                long end = System.currentTimeMillis();
-                elapsed = (end - start) / 1000F;
             }
+        }
+    }
 
-            i++;
-            int percentage = i * 100 / files.size();
-            System.out.println("Extraction time: " + Math.round(elapsed) + "s");
-            System.out.println("Progress: " + percentage + "% done (" + i + " out of " + files.size() + ")");
-            System.out.println("");
+    public static void main(String[] args) throws ParseException, AnalysisException, URISyntaxException, IOException, TransformationException {
+        CommandLineOptionsParser parser = new CommandLineOptionsParser();
+        String error = parser.parse(args);
+        if (error != null) {
+            System.err.println(error + "\n");
+            System.err.println(
+                    "Usage: ContentExtractor -path <path> [optional parameters]\n\n"
+                    + "Tool for extracting metadata and content from PDF files.\n\n"
+                    + "Arguments:\n"
+                    + "  -path <path>           path to a directory containing PDF files\n"
+                    + "  -limit <int>            (optional) the number of PDF files to parse starting from the top  default: \"all\"\n"
```

Comment:
These lines should be kept under 80 characters long, so that the help prints well in a terminal.